src/main/java/org/springframework/data/couchbase/core/ReactiveInsertByIdOperationSupport.java

```diff
@@ -98,7 +97,7 @@ private InsertOptions buildInsertOptions() {
 			} else if (durabilityLevel != DurabilityLevel.NONE) {
 				options.durability(durabilityLevel);
 			}
-			if (expiry != null) {
+			if (expiry != null && ! expiry.isZero()) {
```

Comment:
is this unrelated?

Reply:
yes - unrelated. It's actually in the pr for datacouch-616.